interface/src/app/sample.service.js

```diff
@@ -11,6 +11,35 @@ angular.module('adage.sample.service', [
       // then, can change "limit" below to something sensible
       {limit: 0},
       {
+        'get': {
+          method: 'GET',
+          transformResponse: function(data, headersGetter, status) {
+            var getAnnotation = function(typename) {
+              // We are going to attach getAnnotation to an object, so use of
+              // 'this' below is intentional: we are making a "safe" accessor
+              // method that will return a blank string instead of undefined
+              if (this.annotations !== undefined &&
+                  this.annotations.hasOwnProperty(typename)) {
+                return this.annotations[typename];
+              } else {
+                return '';
+              }
+            };
+            if (status === 200) {
+              data = angular.fromJson(data);
+              if (data.hasOwnProperty('objects')) {
+                // data has a pagination wrapper, so must iterate over list
+                data.objects.forEach(function(obj) {
+                  obj.getAnnotation = getAnnotation;
+                });
+              } else {
```

Comment:
In `GET` method, doesn't `data` always have `objects` property (even when there is only one record) unless `isArray` is set to `true`?
https://docs.angularjs.org/api/ngResource/service/$resource#$resource-returns

Reply:
Great question. The answer turns out to be no, and I included this logic to handle two separate cases that were covered by our existing tests so they would keep working.

There are at least two ways to get a response without an `objects` property (like the one from setting `isArray: true` in the `$resource` definition). Using `isArray` is one, of course. The other way is to make a direct request for a REST "detail" endpoint rather than a "list" endpoint. In other words, we can GET a specific object rather than a list.

The difference is detailed well in these examples (see the sample responses) from the Tastypie documentation:

> **list endpoint**: http://django-tastypie.readthedocs.io/en/latest/interacting.html?highlight=detail#getting-a-collection-of-resources

> **detail endpoint**: http://django-tastypie.readthedocs.io/en/latest/interacting.html?highlight=detail#getting-a-detail-resource

Reply:
Got it. Thanks.